### PR TITLE
fix(ht-gauge-list): added optional input for total

### PR DESCRIPTION
**Pull request by 23nobody**
## Description
Currently, `ht-gauge-list` sets percentage and bar values by default using calculated values. Added support to provide `total` to be used for these calculations.

### Testing
Added test for the two cases of percentages.

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/observability/src/shared/components/gauge-list/gauge-list.component.ts [hunk 1]
```diff
@@ -82,7 +88,7 @@ export class GaugeListComponent<T extends GaugeItem = GaugeItem> implements OnCh
     this.itemOptions = this.items.map(option => ({
       label: option.label,
       color: colorLookupMap.get(option.colorKey ?? option.label),
-      width: `${((option.value / maxValue!) * 100).toFixed(2)}%`,
+      width: `${((option.value / this.total!) * 100).toFixed(2)}%`,
```
Comment (23nobody) [inline projects/observability/src/shared/components/gauge-list/gauge-list.component.ts:91]: Now that we have total being supplied from outside, that total needs to be used for percentage calculation in L93 below.
Thoughts?

File: projects/observability/src/shared/components/gauge-list/gauge-list.component.ts [hunk 2]
```diff
@@ -71,22 +74,35 @@ export class GaugeListComponent<T extends GaugeItem = GaugeItem> implements OnCh
...
-    }));
+
+    if (this.total !== undefined && this.total !== 0) {
+      this.itemOptions = this.items.map(option => ({
+        label: option.label,
+        color: colorLookupMap.get(option.colorKey ?? option.label),
+        width: `${((option.value / this.total!) * 100).toFixed(2)}%`,
+        value: option.value,
+        percentage: (option.value / this.total!) * 100,
+        original: option
```
Comment (itssharmasandeep) [inline projects/observability/src/shared/components/gauge-list/gauge-list.component.ts:86]: This looks duplicate, same as L97 to L103

Comment (itssharmasandeep) [reply projects/observability/src/shared/components/gauge-list/gauge-list.component.ts:86]: can we create a function for those and pass the values as arguments

**Comment by arjunlalb** [comment]
Could you please describe what the PR is intending to do?

File: projects/observability/src/shared/components/gauge-list/gauge-list.component.ts (hunk 2 above)
Comment (23nobody) [reply projects/observability/src/shared/components/gauge-list/gauge-list.component.ts]: Updated. Thanks!

**Comment by arjunlalb** [comment]
Why do we want to use total here?

When there are more than 2 or 3 items, the visualization of the bar doesn't add much value to the user if the fill portion is calculated as a percentage of the total.

Ex:
If there are 5 items to show, each with values 21, 22, 23, 20, 14, all the visual bars would look sort of similar since they are all plotted against the series total of 100.
Also, if the data set contains more than 5 items and we show only 5 in the guage list, the total will be even bigger and will capture no useful information.

We should re-evaluate this.

**Comment by 23nobody** [comment]
> Why do we want to use total here?
> 
> When there are more than 2 or 3 items, the visualization of the bar doesn't add much value to the user if the fill portion is calculated as a percentage of the total.
> 
> Ex: If there are 5 items to show, each with values 21, 22, 23, 20, 14, all the visual bars would look sort of similar since they are all plotted against the series total of 100. Also, if the data set contains more than 5 items and we show only 5 in the guage list, the total will be even bigger and will capture no useful information.
> 
> We should re-evaluate this.

Doesn't that give wrong impression of the distribution, that current 5 items that are in the list contribute in larger extent to the data. 
Showing some items to have more share than the actual is also saying that  the `rest` items comprise lesser shares in the distribution.

**Comment by aaron-steinfeld** [comment]
> > Why do we want to use total here?
> > When there are more than 2 or 3 items, the visualization of the bar doesn't add much value to the user if the fill portion is calculated as a percentage of the total.
> > Ex: If there are 5 items to show, each with values 21, 22, 23, 20, 14, all the visual bars would look sort of similar since they are all plotted against the series total of 100. Also, if the data set contains more than 5 items and we show only 5 in the guage list, the total will be even bigger and will capture no useful information.
> > We should re-evaluate this.
> 
> Doesn't that give wrong impression of the distribution, that current 5 items that are in the list contribute in larger extent to the data. Showing some items to have more share than the actual is also saying that the `rest` items comprise lesser shares in the distribution.

So it sounds like there's two cases 
- Showing distribution of one total - how much does each element contribute to it (e.g. calls to API A as a percentage of all API calls) 
- Showing relative size of related values - how large is each element relative to the largest

We used to exclusively use this for the second case, because the former has much better representations (like a donut, or a single gauge with multiple series which I think we also built). 

If we're trying to extend it to the first case, I would agree we should reconsider the use case and if this is the most appropriate visualization.

File: projects/observability/src/shared/components/gauge-list/gauge-list.component.ts [hunk 3]
```diff
@@ -71,18 +74,28 @@ export class GaugeListComponent<T extends GaugeItem = GaugeItem> implements OnCh
       return;
     }
 
-    let maxValue = maxBy(this.items, option => option.value)?.value;
-    if (maxValue === undefined || maxValue === 0) {
-      maxValue = 1;
+    if (this.total !== undefined && this.total !== 0) {
+      this.setItemOptions(this.items, this.total, this.total);
```
Comment (aaron-steinfeld) [inline projects/observability/src/shared/components/gauge-list/gauge-list.component.ts:78]: There's two different values here - totalCount looks like it got added later (not sure I agree with that from a visualization user case perspective, but a bit besides the point), and it represents that total that would be passed in here. FullCount is the visual indicator, representing the max value.

So I think there's a decent argument to be made to fix those percentages by passing the accurate total from the parent, but the bars should still be based on max value. Once we change that, we're really changing the whole purpose of the visualization.

**Comment by 23nobody** [comment]
Wasn't the correct choice for visualization of the case.

**Pull request closed without merging**